File: Soe.Common/DTO/SearchSysLogsDTO.cs
using SoftOne.Soe.Common.Attributes;
using System;

namespace SoftOne.Soe.Common.DTO
{
    [TSInclude]
    public class SearchSysLogsDTO
    {
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }
        public DateTime? FromTime { get; set; }
        public DateTime? ToTime { get; set; }
        public string Level { get; set; }
        public string LicenseSearch { get; set; }
        public string CompanySearch { get; set; }
        public string RoleSearch { get; set; }
        public string UserSearch { get; set; }
        public string IncMessageSearch { get; set; }
        public string ExlMessageSearch { get; set; }
        public string IncExceptionSearch { get; set; }
        public string ExExceptionSearch { get; set; }
        public int? NoOfRecords { get; set; }
        public bool ShowUnique { get; set; }
    }
}

File: Soe.Angular/TypeScript/Common/Models/SearchSysLogsDTO.ts
import { ISearchSysLogsDTO } from "../../Scripts/TypeLite.Net4";
import { CalendarUtility } from "../../Util/CalendarUtility";

export class SearchSysLogsDTO implements ISearchSysLogsDTO {
	companySearch: string;
	exExceptionSearch: string;
	exlMessageSearch: string;
	fromDate: Date;
	fromTime: Date;
	incExceptionSearch: string;
	incMessageSearch: string;
	level: string;
	licenseSearch: string;
	noOfRecords: number;
	roleSearch: string;
	toDate: Date;
	toTime: Date;
	userSearch: string;
	showUnique: boolean;

	public fixDates() {
		this.fromDate = CalendarUtility.convertToDate(this.fromDate);
		this.toDate = CalendarUtility.convertToDate(this.toDate);
		this.fromTime = CalendarUtility.convertToDate(this.fromTime);
		this.toTime = CalendarUtility.convertToDate(this.toTime);
	}

}


